src/testFixtures/java/org/opensearch/knn/TestUtils.java

```diff
@@ -181,6 +198,35 @@ public static double calculateRecallValue(List<List<String>> searchResults, List
         return sum / recalls.size();
     }
 
+    public static PriorityQueue<DistVector> computeGroundTruthValues(
```

Comment:
Instead of recomputing the index vector, I would consider refactoring to use a class for the index vectors so that logic can be abstracted out. Something like 
```
interface IndexVectorReader {
  int getVectorCount();
  float[] getVector(int id);
}
```

Then, you can have different implementations for getting vector based on ID.